components/omnibox/browser/brave_search_provider.cc

```diff
@@ -6,11 +6,45 @@
 #include "brave/components/omnibox/browser/brave_search_provider.h"
 
 #include "brave/components/omnibox/browser/brave_omnibox_prefs.h"
+#include "build/build_config.h"
 #include "components/omnibox/browser/autocomplete_provider.h"
 #include "components/omnibox/browser/autocomplete_provider_client.h"
+#include "components/omnibox/browser/omnibox_view.h"
 #include "components/omnibox/browser/search_provider.h"
 #include "components/prefs/pref_service.h"
 
+#if !BUILDFLAG(IS_IOS)
+// Got link error on IOS.
+#include "ui/base/clipboard/clipboard.h"
+#include "ui/base/data_transfer_policy/data_transfer_endpoint.h"
+#endif
+
+namespace {
+
+#if !BUILDFLAG(IS_IOS)
+// Copied from chrome/browser/ui/omnibox/clipboard_utils.h
+std::u16string GetClipboardText() {
```

Comment:
pls implement this as a part of `AutocompleteProviderClient`. no need to pull the function into `components/`.